src/helper/helper.js

```diff
@@ -48,165 +48,157 @@ goog.module('helper');
 
 const {type, hasOwn, isPlainObject} = goog.require('plain');
 
-
 /**
- * Creates a new helper object for the given dataLayer.
+ * A helper that will listen for new messages on the given dataLayer.
+ * Each new message will be merged into the helper's "abstract data model".
+ * This internal model object holds the most recent value for all keys which
+ * have been set on messages processed by the helper.
  *
- * @constructor
- * @param {!Array.<!Object>} dataLayer The dataLayer to help with.
- * @param {function(!Object, !Object)=} opt_listener The callback function to
- *     execute when a new state gets pushed onto the dataLayer.
- * @param {boolean=} opt_listenToPast If true, the given listener will be
- *     executed for state changes that have already happened.
+ * You can retrieve values from the data model by using the helper's 'get' method.
  */
-const DataLayerHelper = function(dataLayer, opt_listener, opt_listenToPast) {
-
+class DataLayerHelper {
   /**
-   * The dataLayer to help with.
-   * @type {!Array.<!Object>}
-   * @private
-   */
-  this.dataLayer_ = dataLayer;
-
-  /**
-   * The listener to notify of changes to the dataLayer.
-   * @type {function(!Object, !Object)}
-   * @private
+   * Creates a new helper object for the given dataLayer.
+   *
+   * @param {!Array.<!Object>} dataLayer The dataLayer to help with.
+   * @param {function(!Object, !Object)=} listener The callback function to
+   *     execute when a new state gets pushed onto the dataLayer.
+   * @param {boolean=} listenToPast If true, the given listener will be
+   *     executed for state changes that have already happened.
    */
-  this.listener_ = opt_listener || function() {};
-
-  /**
-   * The internal marker for checking if the listener is currently on the stack.
-   * @type {boolean}
-   * @private
-   */
-  this.executingListener_ = false;
+  constructor(dataLayer, listener, listenToPast) {
+    /**
+     * The dataLayer to help with.
+     * @private @const {!Array.<!Object>}
+     */
+    this.dataLayer_ = dataLayer;
+
+    /**
+     * The listener to notify of changes to the dataLayer.
+     * @private @const {function(!Object, !Object)}
+     */
+    this.listener_ = listener || function() {};
+
+    /**
+     * The internal marker for checking if the listener is currently on the stack.
+     * @private @const {boolean}
```

Comment:
This one is not @const because it gets changed down in L194. Can you please re-audit these?

Reply:
Sorry about that - got a bit @const happy 🤦 

There is a compiler warning (not error) so I can say with confidence this is the only incorrectly declared const. Alex is working on updating the build to fail rather than error. Thanks for catching this!